packages/@aws-cdk/aws-events/lib/event-bus.ts

```diff
@@ -96,12 +115,50 @@ export interface EventBusAttributes {
   readonly eventSourceName?: string;
 }
 
+abstract class EventBusBase extends Resource implements IEventBus {
+  /**
+   * The physical ID of this event bus resource
+   */
+  public abstract readonly eventBusName: string;
+
+  /**
+   * The ARN of the event bus, such as:
+   * arn:aws:events:us-east-2:123456789012:event-bus/aws.partner/PartnerName/acct1/repo1.
+   */
+  public abstract readonly eventBusArn: string;
+
+  /**
+   * The policy for the event bus in JSON form.
+   */
+  public abstract readonly eventBusPolicy: string;
+
+  /**
+   * The name of the partner event source
+   */
+  public abstract readonly eventSourceName?: string;
+
+  public archive(
```

Comment:
The typical form of these methods is to take an `id, props` pair as arguments.

Definitely I'd want the props type, a list of positional arguments is not properly extensible and becomes very unfriendly quickly.

Reply:
That makes sense. I should have done that from the start.